src/vs/workbench/contrib/search/browser/searchView.ts

```diff
@@ -1043,6 +1043,37 @@ export class SearchView extends ViewletPanel {
 		this.onQueryChanged(true);
 	}
 
+	searchReplace(query?: string, replace?: string, triggerSearch?: boolean, caseSensitive?: boolean, wholeWords?: boolean, regex?: boolean, filesToInclude?: string, filesToExclude?: string): void {
+		if (typeof caseSensitive === 'boolean') {
+			this.searchWidget.searchInput.setCaseSensitive(caseSensitive);
+		}
+		if (typeof wholeWords === 'boolean') {
+			this.searchWidget.searchInput.setWholeWords(wholeWords);
+		}
+		if (typeof regex === 'boolean') {
+			this.searchWidget.searchInput.setRegex(regex);
+		}
+		if (typeof filesToInclude === 'string') {
+			this.searchIncludePattern.setValue(String(filesToInclude));
+		}
+		if (typeof filesToExclude === 'string') {
+			this.searchExcludePattern.setValue(String(filesToExclude));
+		}
+		if (typeof query === 'string') {
+			this.searchWidget.searchInput.setValue(query);
+		}
+		if (typeof replace === 'string') {
+			this.searchWidget.replaceInput.value = replace;
+		} else {
+			if (this.searchWidget.replaceInput.value !== '') {
+				this.searchWidget.replaceInput.value = '';
+			}
+		}
+		if (typeof triggerSearch === 'boolean' && triggerSearch) {
+			this.onQueryChanged(true);
```

Comment:
To be honest, I have to think about whether we want extensions to have the ability to actually trigger the search. It kind of feels like a strange thing for extensions to drive the UI.

Reply:
Oh, I thought this was one of the major points with this issue.

Actually you can already trigger the search today, all you need to do is to call the toggleRegex(), then a search is performed... So having this parameter doesn't really change anything.